### PR TITLE
Feature/localstorage

### PR DESCRIPTION
**Pull request by colefiscus**
#### What does  this PR do?

Added functionality for local storage, favoriting an idea, deleting an idea, and searching for an idea.

When a user adds an idea, the properties of that idea will be stored in local storage. When a user deletes an idea, that idea will be removed from local storage. When a user uses the search function, only ideas that include the phrase or word in the input will be shown.

Small fixes to parameters and other names.

#### Where should the reviewer start?

The main section to review is from lines 103 - 143. Also ensure small, various edits to CSS and functions make sense. 

#### How should this be manually tested?

Multiple ideas should be added, one should be favorited. On page refresh, both ideas should be present with appropriate favoriting. The search function should be used to make sure only desired cards appear. When nothing is present inside search tool, all cards should be visible.

When a card is deleted, the card should not reappear on page refresh.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -71,20 +77,68 @@ function checkEventTarget(event) {
...
+        var idea = JSON.parse(localStorage.getItem(id));
+        idea.star = !idea.star;
+        localStorage.setItem(id, JSON.stringify(idea));
       };
     };
-    showIdeas()
+    showIdeas(savedIdeas)
 };
+
+function pageLoad() {
```
Comment (JeffKersting) [inline src/main.js:103]: Add a sort function here, to display ideas in order in which created.

File: src/idea.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
 class Idea {
-  constructor(title, body){
-    this.id = Date.now();
+  constructor(title, body, id, star){
```
Comment (DrewBradley) [inline src/idea.js:2]: This is such a clear way to allow us to create new instances while reloading from local storage.

File: src/main.js [hunk 3]
```diff
@@ -8,23 +8,29 @@ var searchInput = document.querySelector(".search-input");
...
 var ideaGrid = document.querySelector(".idea-grid");
 
+// var retrievedObject = JSON.parse(localStorage.getItem("localIdeas"));
 var savedIdeas = [];
 
 // EVENT LISTENERS
 saveButton.addEventListener('click', saveIdea);
 bodyInput.addEventListener('keyup', enableSaveButton);
 ideaGrid.addEventListener("click", checkEventTarget);
+window.addEventListener('load', pageLoad);
```
Comment (DrewBradley) [inline src/main.js:18]: This is such concise naming!

File: src/main.js [hunk 4]
```diff
@@ -71,20 +77,68 @@ function checkEventTarget(event) {
...
+  if (showStarredButton.innerText === "Show Starred Ideas"){
+    showStarredButton.innerText = "Show All Ideas"
+    filterStarredIdeas(savedIdeas);
+  }else{
+    showStarredButton.innerText = "Show Starred Ideas"
+    showIdeas(savedIdeas);
+  }
+}
+
+function filterStarredIdeas(array){
```
Comment (DrewBradley) [inline src/main.js:127]: Is this function doing the same thing as line 134? It seems to be achieving the same outcome.

File: styles.css [hunk 5]
```diff
@@ -96,6 +96,12 @@ button {
   font-size: x-large;
   font-family: Open Sans;
   font-weight: 700;
+  cursor: pointer;
+}
+button:disabled{
+  background-color: #6b6c88;
+  color: #555570;
+  cursor: not-allowed;
```
Comment (DrewBradley) [inline styles.css:104]: Have we made sure this cursor is showing up correctly?

**Review by DrewBradley — APPROVED** [review]

**Pull request merged**